apps/crm/src/modules/mailchimp/mailchimp-list-picker/mailchimp-list-picker.tsx

```diff
@@ -0,0 +1,39 @@
+import { Box, PropsWithBox } from "@saleor/macaw-ui/next";
+import { ChangeEvent, useEffect } from "react";
+import { trpcClient } from "../../trpc/trpc-client";
+
+type Props = PropsWithBox<{
+  disabled?: boolean;
+  onChange(e: ChangeEvent<HTMLSelectElement> | null, listId: string): void;
+}>;
+
+export const MailchimpListPicker = ({ disabled, onChange, ...props }: Props) => {
+  const { isSuccess, data, isLoading } = trpcClient.mailchimp.audience.getLists.useQuery();
+
+  if (isSuccess && !data) {
+    console.error("Fetched empty audiences list, should not happen");
+  }
+
+  useEffect(() => {
+    if (data?.length) {
```

Comment:
Could it be moved to react-query `onSuccess` hook? I find such approach easier to follow than useEffect pattern.